app/views/search/results/_protocols.html.erb

```diff
@@ -48,6 +48,47 @@
       <%=l protocol.created_at, format: :full %>
     </span>
     <br>
+    <% if protocol.nr_of_linked_children > 0 %>
+      <span>
+        <% if protocol.linked_modules %>
+          <%=t "search.index.modules" %>
+          <% protocol.linked_modules.each_with_index do |mod, i| %>
+            <%= render partial: "search/results/partials/my_module_text.html.erb",
+                       locals: { my_module: mod } %>
+            <% if i != (sub_results(protocol.linked_modules).count - 1) %>
+              ,&nbsp;
+            <% end %>
+          <% end %>
+        <% end %>
+      </span>
+      <br>
+      <span>
+        <% if protocol.linked_experiments %>
```

Comment:
I'd say you'd rather save `linked_modules` for the first time and then pass it to the `linked_experiments`. Now you do 2 queries for this. And 3 for projects below, whereas it could be 1 for every entity.